Update Docs Homepage

gideongrinberg opened the pull request:
I'm working on updating the docs homepage to be a bit more concise and readable. I also updated the theme to Alabaster. I have to go, but I will update this with a full list of changes in a little while. This PR closes #409

gideongrinberg committed:
First pass at better homepage

gideongrinberg commented:
I still need to update it based on these specs:

> Currently, it's a stale copy of the README. As part of an attempt to improve the landing pages for various users, we should rewrite it.
> 
> The kinds of people who will be coming to our docs landing page (and caring about it) fall into two categories:
> 
> People who are directed there through one of our messaging platforms (ppb.dev, the twitter, or linked directly from inside the discord)
> experienced developers who prefer going straight to the documentation.
> As such, this needs to be able to assist these two groups as quickly as possible.
> 
> Functional outline:
> 
> Introduction: what is ppb, what problem is it trying to solve.
> The structure of the documentation. Tell our users where the thing they might want to know is. (reference https://www.divio.com/blog/documentation/ for further details. Our documentation layout is based on this model.)
> The next focus should be new users: Link directly to the getting started guide, explain what assumptions that guide has (you probably need to know how to program to really use our getting started guide.)
> After that, focus on more experiences folks. They're going to be most interested in the cookbook and the API documentation. They want to solve specific problems. Explain how those are laid out, provide links.
> At the end, standard TOC.

pathunstrom commented:
I'm concerned about the raw HTML block. Is there a reason rst content didn't work?

gideongrinberg commented:
This is a rough draft, and I'm new to RST so I will fix that.

AstraLuma reviewed (commented):
I'm going to mostly let you finish before getting into reviewing this proper, but I want to give you some guidance before you put in too much work.

 One of the things we want to do is specialize the website, the docs frontpage, and the README to be more focused on what they're doing, instead of being all copies of each other. As it stands, the text here seems to focus more on promoting and "selling" ppb, which is the job of the website. If the reader is looking at the docs, they're already "sold" and we don't need to be sold again. IMHO, this page should be focusing more on either architectural/technical principles or info about the documentation itself.

gideongrinberg commented:
Thank you, I'll keep that in mind, although sometimes I do look over the docs of libraries to assess them.

gideongrinberg committed:
Restructure and add content

gideongrinberg commented:
@AstraLuma thanks for the feedback. I've updated the docs. The basic structure is now:

- [x] Heading / Intro (Adapted from README)
- [ ] Using these docs (Summary of structure, needs expansion. 
  - New User, explains tutorials/guide, links to python tutorial (Should I link to the old text about PPB?)
  - Experienced User, links to reference and cookbooks (Todo: fix links)
- [x] Toctree

gideongrinberg committed:
Add admonition, fix logo

gideongrinberg committed:
Delete what.html

gideongrinberg committed:
Fix links. This should work, but not sure.

pathunstrom commented on docs/index.rst:
```diff
@@ -1,83 +1,29 @@
-discovery are both emphasized. A new user should be able to build their
-first game in a few hours, and continue exploring beyond that.
+As an Experienced User
+^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Built locally, rst didn't like the line length of the underline. Best practice is to overkill since when we start adding translations, this line must be long enough for all translations.

pathunstrom commented on docs/index.rst:
```diff
@@ -1,83 +1,29 @@
-and your handlers should match the signature
-``on_update(self, update_event, signal)``.
+As a New User
+^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Built locally, rst didn't like the line length of the underline. Best practice is to overkill since when we start adding translations, this line must be long enough for all translations.

pathunstrom commented on docs/index.rst:
```diff
@@ -1,83 +1,29 @@
 
-A Game Engine
--------------
+.. note :: This document explains the practical and technical details of :code:`ppb`. If you want to know more about what it is, its goals, or why you should use it, check out `this page<https://github.com/ppb/pursuedpybear/blob/canon/README.md>` `our website<https://ppb.dev>`.
```
```suggestion
.. note :: This document explains the practical and technical details of :code:`ppb`. If you want to know more about what it is, its goals, or why you should use it, check out `this page<https://github.com/ppb/pursuedpybear/blob/canon/README.md>`_ of `our website<https://ppb.dev>`_.
```

pathunstrom commented on docs/index.rst:
```diff
@@ -1,83 +1,29 @@
-and UI screens in a simple way.
+Using these docs
+----------------
+Our documentation is seperated into practical steps and theroretical knowledge. The docs are made up of Tutorials, which allow you to learn quickly, and guides which bring you to a specific goal. We also have a standard API reference.
```
```suggestion
Our documentation is separated into practical steps and theoretical knowledge. The docs are made up of Tutorials, which allow you to learn quickly, and guides which bring you to a specific goal. We also have a standard API reference.
```

pathunstrom reviewed (changes requested):
All right, these should be the last comments.

Will open an outside issue regarding the theme choice, since no decision had been reached on it.

gideongrinberg committed:
Fix heading length

Based on @pathunstrom's suggestion, updated line length to make the linter happy and allow for translations

Co-authored-by: Piper Thunstrom <pathunstrom@gmail.com>

gideongrinberg committed:
Fix heading length

Co-authored-by: Piper Thunstrom <pathunstrom@gmail.com>

gideongrinberg committed:
Fix links

Co-authored-by: Piper Thunstrom <pathunstrom@gmail.com>

gideongrinberg committed:
Fix spelling mistakes

Co-authored-by: Piper Thunstrom <pathunstrom@gmail.com>

gideongrinberg committed:
Remove theme

gideongrinberg committed:
Delete unused CSS

gideongrinberg reviewed (commented):
I believe this is will now be ready once the logo file is deleted. I apologize for the rather silly mistakes in it.

pathunstrom commented:
No worries about the mistakes, we all make them. Trying to figure out why the build is failing. Will get back to you about it soon.

gideongrinberg commented:
It seems to be an issue with my links

```
/tmp/cirrus-ci-build/docs/index.rst:6: WARNING: Unknown target name: "this page<https://github.com/ppb/pursuedpybear/blob/canon/readme.md>".
/tmp/cirrus-ci-build/docs/index.rst:6: WARNING: Unknown target name: "our website<https://ppb.dev>".
```
I thought I did it right, but I'll review the docs.

gideongrinberg committed:
Fix links, change "of" to "or"

For some odd reason or another, our CI build was failing do to a hyperlink issue. I changed our syntax a bit, so it should be all good now. I screwed up my local Sphinx so this might not work the same way it did on my system.

pathunstrom commented:
The actual error that the docs build is throwing is regards to the logo option.

We should check which version of alabaster is being installed and make sure we're looking at the correct docs. Alternatively, we can revert the alabaster change for now and remove those options, and use the sphinx standard [html_logo](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_logo) configuration instead.

gideongrinberg commented:
That would be the issue, there is no longer a logo option.  Let's just get rid of alabaster

pathunstrom commented:
Definitely join the conversation in the linked issue about docs styling if you're interested in what we change to.

gideongrinberg committed:
Update conf.py

pathunstrom reviewed (approved):
Everything looks good enough.

Thanks for the contribution!

bors r+

pathunstrom commented:
Actually, noticed one thing: You didn't add your name to the CONTRIBUTORS.md!

Don't worry about it for this one. Feel free to add another PR with that change on its own.

gideongrinberg commented:
> Actually, noticed one thing: You didn't add your name to the CONTRIBUTORS.md!
> 
> Don't worry about it for this one. Feel free to add another PR with that change on its own.

Thanks @pathunstrom
